File: sdbg/src/main/java/org/cf/sdbg/command/Where.java
package org.cf.sdbg.command;

import org.cf.sdbg.Main;
import org.cf.smalivm.context.ExecutionNode;
import picocli.CommandLine;

@CommandLine.Command(name = "where", aliases = {"w"}, mixinStandardHelpOptions = true, version = "1.0",
    description = "Shows current instruction index and which function you are in")
public class Where implements Runnable {

  @CommandLine.ParentCommand
  CliCommands parent;

  public void run() {
    int instructionIndex = Main.debugger.getCurrentIndex() + 1;
    String methodSignature = Main.debugger.getCurrentMethodSignature();

    parent.out.println("#" + instructionIndex + " in " + methodSignature);
  }
}

File: sdbg/src/main/java/org/cf/sdbg/command/Where.kt
package org.cf.sdbg.command

import org.cf.sdbg.Main
import picocli.CommandLine
import picocli.CommandLine.ParentCommand

@CommandLine.Command(name = "where", aliases = ["w"], mixinStandardHelpOptions = true, version = ["1.0"], description = ["Shows current instruction index and which function you are in"])
class Where : Runnable {
    @ParentCommand
    var parent: CliCommands? = null

    override fun run() {
        val instructionIndex = Main.debugger!!.currentIndex + 1
        val methodSignature = Main.debugger!!.currentMethodSignature
        parent!!.out!!.println("#$instructionIndex in $methodSignature")
    }
}
